File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi=nums[0]
        res=nums[0]

        for i in range(1,len(nums)):
            maxi=max(maxi+nums[i],nums[i])
            res=max(res,maxi)

        return res
        

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi=INT_MIN;
        int curr=0;

        for(auto num:nums){
            curr=max(curr+num,num);
            maxi=max(curr,maxi);

        }

        return maxi;
    }
};